cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -40,6 +42,100 @@ pub struct ContractAliasInput {
     pub new_alias: Option<String>,
 }
 
+pub fn propose_pfm(
+    ctx: (DepsMut, Env, MessageInfo),
+    chain: String,
+) -> Result<Response, ContractError> {
+    let (deps, env, info) = ctx;
+
+    // enforce lowercase
+    let chain = chain.to_lowercase();
+
+    // validation
+    let registry = Registry::default(deps.as_ref());
+    let coin = cw_utils::one_coin(&info)?;
+    let native_chain = registry.get_native_chain(&coin.denom)?;
+
+    if native_chain.as_ref() != chain {
+        return Err(ContractError::CoinFronInvalidChain {
+            supplied_chain: native_chain.as_ref().to_string(),
+            expected_chain: chain,
+        });
+    }
+
+    // check if the chain is already registered or is in progress
+    if let Some(chain_pfm) = CHAIN_PFM_MAP.may_load(deps.storage, &chain)? {
+        if chain_pfm.is_validated() {
+            // Only authorized addresses can ask for a validated PFM to be re-checked
+            // If sender is the contract governor, then they are authorized to do do this to any chain
+            // Otherwise, they must be authorized to do manage the chain they are attempting to modify
+            let user_permission =
+                check_is_authorized(deps.as_ref(), info.sender, Some(chain.clone()))?;
+            check_action_permission(FullOperation::Change, user_permission)?;
+        } else {
+            return Err(ContractError::PFMValidationAlreadyInProgress {
+                chain: chain.clone(),
+            });
+        }
+    };
+
+    // Store the chain to validate
+    CHAIN_PFM_MAP.save(deps.storage, &chain, &ChainPFM::default())?;
+
+    let own_addr = env.contract.address;
+
+    // redeclaring (shadowing) registry to avoid issues with the borrow checker
+    let registry = Registry::default(deps.as_ref());
+    let ibc_transfer = registry.unwrap_coin_into(
+        coin,
+        own_addr.to_string(),
+        None,
+        own_addr.to_string(),
+        env.block.time,
+        format!(r#"{{"ibc_callback":"{}"}}"#, own_addr),
+        Some(Callback {
+            contract: own_addr,
+            msg: format!(r#"{{"validate_pfm": {{"chain": "{}"}} }}"#, chain).try_into()?,
+        }),
+        true,
+    )?;
```

Comment:
Is my understanding correct that we send an ibc transfer and on successful completion, a wasm hook is called that runs `validate_pfm`?

Reply:
yes! correct